src/node_crypto.cc

```diff
@@ -255,13 +255,50 @@ void ThrowCryptoError(Environment* env,
                       unsigned long err,  // NOLINT(runtime/int)
                       const char* default_message = nullptr) {
   HandleScope scope(env->isolate());
+  Local<String> message;
+
   if (err != 0 || default_message == nullptr) {
     char errmsg[128] = { 0 };
     ERR_error_string_n(err, errmsg, sizeof(errmsg));
-    env->ThrowError(errmsg);
+    message = String::NewFromUtf8(env->isolate(), errmsg,
+                                  v8::NewStringType::kNormal)
+                                      .ToLocalChecked();
   } else {
-    env->ThrowError(default_message);
+    message = String::NewFromUtf8(env->isolate(), default_message,
+                                  v8::NewStringType::kNormal)
+                                      .ToLocalChecked();
+  }
+
+  Local<Value> exception_v = Exception::Error(message);
+  CHECK(!exception_v.IsEmpty());
+  Local<Object> exception = exception_v.As<Object>();
+  Local<Array> error_stack = Array::New(env->isolate());
+
+  ERR_STATE *es = ERR_get_state();
+  // Build the error_stack array to be added to openSSLErrorStack property.
+  for (unsigned int i = 0; es->bottom != es->top
+        && (es->err_flags[es->top] & ERR_FLAG_MARK) == 0; i++) {
```

Comment:
This doesn't look correct to me.  For starters, `es->top` and `es->bottom` should be restored after you're done.  Second, this assumes `es->bottom <= es->top` but that isn't necessarily true because it's a ring buffer.

Reply:
ok. i think i understand. I'll fix this